internal/lsp/client.go

```diff
@@ -470,3 +479,18 @@ func HasRootMarkers(dir string, rootMarkers []string) bool {
 	}
 	return false
 }
+
+// isFileInWorkingDir checks if the given file path is within the client's working directory.
+func (c *Client) isFileInWorkingDir(filePath string) bool {
+	absFilepath, err := filepath.Abs(filePath)
+	if err != nil {
+		return false
+	}
+
+	absWd, err := filepath.Abs(c.workingDir)
+	if err != nil {
+		return false
+	}
+
+	return strings.HasPrefix(absFilepath, absWd)
```

Comment:
Oh, just noticed that we have already a `fsext.HasPrefix` helper, so no need for extra helpers.

This PR is probably from before we had it.

Reply:
I actually just pushed a commit with this change 🙂